src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -341,6 +340,16 @@ private void OnHostLeaseChanged(object sender, EventArgs e)
         /// </summary>
         private void PreInitialize()
         {
+            // Log whether App Insights is enabled
+            if (!string.IsNullOrEmpty(_settingsManager.ApplicationInsightsInstrumentationKey))
+            {
+                _metricsLogger.LogEvent(MetricEventNames.ApplicationInsightsEnabled);
+            }
+            else
+            {
+                _metricsLogger.LogEvent(MetricEventNames.ApplicationInsightsDisabled);
+            }
```

Comment:
@fabiocav -- any idea of a better way to do this? When I make the decision, I don't have IMetricsLogger. This was my quick fix to make sure this was logged.

At the very least I'd like to change this to check whether the LoggerProvider is registered (rather than look for the key).

Reply:
Are you talking about a startup service that can inspect other registered services and log, etc?

Reply:
Something like that. We could also have deferred logs... need to think through this.